packages/react-core/src/components/Select/SelectOption.tsx

```diff
@@ -109,16 +121,47 @@ export class SelectOption extends React.Component<SelectOptionProps> {
       index,
       component,
       inputId,
+      isFavorite,
+      ariaIsFavoriteLabel = 'starred',
+      ariaIsNotFavoriteLabel = 'not starred',
       ...props
     } = this.props;
     /* eslint-enable @typescript-eslint/no-unused-vars */
     const Component = component as any;
+
+    if (!id && isFavorite !== null) {
+      // eslint-disable-next-line no-console
+      console.error('Please provide an id to use the favorites feature.');
+    }
+
+    const generatedId = id || getUniqueId('select-option');
+    const favoriteButton = (onFavorite: any) => (
+      <button
+        className={css(styles.selectMenuItem, styles.modifiers.action, styles.modifiers.favoriteAction)}
+        aria-label={isFavorite ? ariaIsFavoriteLabel : ariaIsNotFavoriteLabel}
+        onClick={() => {
+          onFavorite(generatedId, isFavorite);
+        }}
+      >
+        <span className={css(styles.selectMenuItemActionIcon)}>
+          <StarIcon />
+        </span>
+      </button>
+    );
+
     return (
       <SelectConsumer>
-        {({ onSelect, onClose, variant, inputIdPrefix }) => (
+        {({ onSelect, onClose, variant, inputIdPrefix, onFavorite }) => (
           <React.Fragment>
             {variant !== SelectVariant.checkbox && (
-              <li role="presentation">
+              <li
+                id={generatedId}
```

Comment:
is this id here needed? will fail snapshot tests

Reply:
Yes.  It is needed.  Previously the id was generated by the menu, I moved her to facilitate favorites.  The random Id generator was not being used previously so the id was predictable and the test cases would pass. So, when a consumer creates an option in the select, the do no supply an id and it is always generated causing snapshot to fail.